src/System.Linq/src/System/Linq/Partition.cs

```diff
@@ -42,22 +42,18 @@ internal interface IPartition<TElement> : IIListProvider<TElement>
 
         IPartition<TElement> Take(int count);
 
-        TElement ElementAt(int index);
+        TElement TryGetElementAt(int index, out bool found);
 
-        TElement ElementAtOrDefault(int index);
+        TElement TryGetFirst(out bool found);
 
-        TElement First();
-
-        TElement FirstOrDefault();
-
-        TElement Last();
-
-        TElement LastOrDefault();
+        TElement TryGetLast(out bool found);
     }
 
     internal sealed class EmptyPartition<TElement> : IPartition<TElement>, IEnumerator<TElement>
     {
-        public EmptyPartition()
+        public static readonly IPartition<TElement> Instance = new EmptyPartition<TElement>();
```

Comment:
It'd be hard to get a good measure off, but I'd be prepared to bet it would be a win. However, I'd also be prepared to bet that there is code out there with a dependency on the fact that `Empty` currently returns an array. Indeed prior to `Array.Empty` in 4.6 it was a way to get a singleton array that was reasonably likely to be shared with other assemblies.